src/bitmessageqt/tests/settings.py

```diff
@@ -0,0 +1,34 @@
+import threading
+import time
+
+from main import TestBase
+from bmconfigparser import BMConfigParser
+from bitmessageqt import settings
+
+
+class TestSettings(TestBase):
+    """A test case for the "Settings" dialog"""
+    def setUp(self):
+        super(TestSettings, self).setUp()
+        self.dialog = settings.SettingsDialog(self.window)
+
+    def test_udp(self):
+        """Test the effect of checkBoxUDP"""
+        udp_setting = BMConfigParser().safeGetBoolean(
+            'bitmessagesettings', 'udp')
+        self.assertEqual(udp_setting, self.dialog.checkBoxUDP.isChecked())
+        self.dialog.checkBoxUDP.setChecked(not udp_setting)
+        self.dialog.accept()
+        self.assertEqual(
+            not udp_setting,
+            BMConfigParser().safeGetBoolean('bitmessagesettings', 'udp'))
+        time.sleep(5)
```

Comment:
It's a good practice to avoid `time.sleep` in tests, but unfortunately I don't have a good alternative for the time being.

Reply:
So why `QtTest` has it's own [qSleep()](https://www.riverbankcomputing.com/static/Docs/PyQt5/api/qttest/qtest.html#qWait) if it's considered a bad practice? In general I agree, refactoring of the code and improving tests will take some time.